### ISSUE TITLE
doctest on autofunction does is hidden with certain decorators

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#121]
When using autofunction to include API docs for a function that is decorated _and_ contains a doctest, the doctest does not get run.  However, the function's docstring _is_ included in sphinx documentation as expected.

I know that decorators can adjust function properties but I've looked closely at this particular decorator and I think it is setting enough properties.

To reproduce:
1. create an autofunction:: mod.some_ctx -- that is, one that points to a function with a decorator

I was using the builtin contextmanager like this:

```
#!python
from contextlib import contextmanager
@contextmanager
def some_ctx():
    """this doctest should fail

    >>> 1+1
    3

    """"
    yield None
```
1. run make html for sanity so you can see the API doc for some_ctx()
2. run make doctest

There should be a failure but there is not.  Now, comment out @contextmanager, rerun make doctest and there will be a failure.

Anyway, this was the behavior I saw with Sphinx 0.5.1.  I looked at the source for contextmanager() in 2.5 and could not see anything peculiar.  It does not assign helper.**module** = func.**module** but when I added that as an experiment there was no affect.

My only guess is that this is not getting put into nodes.doctest_block but if that is the case then I don't see how it would be rendered as API docs.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/121
- Originally reported by: Kumar McMillan
- Originally created at: 2009-03-10T05:06:55.790

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2009-03-24 09:04:45+00:00_

Will look at this after 0.6 is released.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2009-05-02 17:44:01+00:00_

This is caused by http://bugs.python.org/issue1108, which should be fixed in some 2.6 bugfix release.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-09-22 10:14:37+00:00_

Removing milestone: 1.0 (automated comment)
